dissect/target/loaders/cb.py

```diff
@@ -1,143 +1,197 @@
-from urllib.parse import urlparse
+from __future__ import annotations
 
-from cbapi.live_response_api import LiveResponseError
-from cbapi.response import CbResponseAPI, Sensor
+from datetime import datetime
+from functools import cached_property
+from pathlib import Path
+from typing import TYPE_CHECKING, Optional
+from urllib.parse import ParseResult
+
+try:
+    from cbc_sdk.errors import CredentialError
+    from cbc_sdk.live_response_api import LiveResponseSession
+    from cbc_sdk.platform import Device
+    from cbc_sdk.rest_api import CBCloudAPI
+except ImportError:
+    raise ImportError("Please install 'carbon-black-cloud-sdk-python' to use the 'cb://' target.")
+
+from dissect.util import ts
 
 from dissect.target.exceptions import (
+    LoaderError,
+    RegistryError,
     RegistryKeyNotFoundError,
     RegistryValueNotFoundError,
 )
 from dissect.target.filesystems.cb import CbFilesystem
 from dissect.target.helpers.regutil import RegistryHive, RegistryKey, RegistryValue
+from dissect.target.helpers.utils import parse_path_uri
 from dissect.target.loader import Loader
 from dissect.target.plugins.os.windows.registry import RegistryPlugin
 
+if TYPE_CHECKING:
+    from dissect.target.target import Target
+
 
 class CbLoader(Loader):
-    def __init__(self, path, **kwargs):
-        path = str(path)
+    def __init__(self, path: str, parsed_path: ParseResult = None, **kwargs):
+        self.host, _, instance = parsed_path.netloc.partition("@")
         super().__init__(path)
-        self.uri = urlparse(path)
-        self.host = self.uri.netloc
 
-        self.cb = CbResponseAPI()
-        if self.host.isdigit():
-            self.sensor = self.cb.select(Sensor, self.host)
-        else:
-            q = self.cb.select(Sensor)
+        # A profile will need to be given as argument to CBCloudAPI
+        # e.g. cb://workstation@instance
```

Comment:
Maybe elaborate this a bit and move it to the top of the file?

I would say:

```
- Usage: cb://<sensorname>@<instance>
- Instance: section in config (for example: default)
- Sensorname: the name of the sensor as listed in the inventory screen
- Add credentials in ~/.carbonblack/credentials.cbc (chmod 600)
```

Something like that?